### ISSUE TITLE
CA2000 not detected when using nameof() inside the method

### ISSUE THREAD

**Issue by tompazourek** [dotnet/roslyn-analyzers#3297]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

Example: [CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000)

#### Repro steps

1. Create an empty solution from the .NET Framework 4.8 Console Application.
2. Add "Microsoft.CodeAnalysis.FxCopAnalyzers" version "2.9.8".
3. Add the following code:

```csharp
    public class Baz
    {
        // doesn't trigger CA2000
        public string Foo()
        {
            var streamReader = new StreamReader(Stream.Null);
            Console.WriteLine(nameof(Foo));
            return streamReader.ReadToEnd();
        }
    
        // triggers CA2000
        public string Bar()
        {
            var streamReader = new StreamReader(Stream.Null);
            Console.WriteLine("Bar");
            return streamReader.ReadToEnd();
        }
    }
```

#### Expected behavior

Both the `Bar` and `Foo` methods get flagged for CA2000 violation.

#### Actual behavior

Only the `Bar` method gets flagged for CA2000 violation.
